Logging/LogHandler/FileLogHandler.cs

```diff
@@ -94,9 +163,54 @@ protected override void DisposeSelf()
             string context = string.Format(
                 LogContextFormat,
                 callerDerivedTypeName, filename, callerName, callerLine
-                );
+            );
 
             m_Writer.WriteLine($"{timestamp}{logLevel}{context}{message}");
+
+            if (m_ShouldRotate && m_Writer.BaseStream.Length > RotateFileSizeLimit)
+            {
+                this.RotateFiles();
+            }
+        }
+
+        private void RotateFiles()
+        {
+            Debug.Assert(RotateFileSizeLimit >= 0, $"Expected positive integer for file size limit, but got {RotateFileSizeLimit}");
+            Debug.Assert(RotateFileCountLimit >= 0, $"Expected positive integer for file count limit, but got {RotateFileCountLimit}");
+
+            m_Writer.Close();
+            m_Writer = null;
+
+            // Determine how many rotated log files already exist
+            int index = 1;
+            string rotatedFilePath = GetRotatedFilePath(index);
```

Comment:
Good idea, I wasn't aware! I'll took at improving that.

Reply:
What should happen when there's a gap? If only 1 and 3 exist, should they become 2 and 3, or 2 and 4?

I'm guessing the former, i.e. don't necessarily rotate every log file, only do so when attempting to rotate a newer file into its place.

Reply:
I'd say cull anything after continuity is broken. Otherwise could lead to some confusing situations where you think two adjacent log files are connected through a session but really aren't.

Of course, you could look at the date/time on the message but it would be an easy oversight.

Reply:
I could do that too - I can still use `Directory.GetFiles()` to pull a list of rotated files, but instead of handling them all I would just step through them sequentially starting at 1, and as soon as I reach a gap, delete all remaining files? Then I could remove all the custom handling I've currently got, which attempts to preserve any files it finds.

Reply:
> I would just step through them sequentially starting at 1, and as soon as I reach a gap, delete all remaining files?

Exactly what I was thinking.

Reply:
Done.